### DISCUSSION TITLE
Possible cache overuse?

### DISCUSSION THREAD

**Discussion by gtg250y** [FreeTubeApp/FreeTube#3118] [General]
I just noticed my hard drive space had shrunk dramatically recently. Tracked it down to FreeTube, which I just always leave open (opening all videos in new windows from one main instance).

In about 3 weeks, this directory had ~55,000 files, each around ~70MB, totaling about ~60GB: AppData\Roaming\FreeTube\Cache\Cache_Data

After closing everything, then upon re-launching the app, it cleared it all out finally.

Does that sound off, or has anyone else seen this? This is on v18.0, and I don't think this was like that as of a version or two ago (or at least I never noticed).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
This sounds like an electron issue and not something wrong on our end.

Could u you try the experimental replace http cache option in the settings. This nukes electron's http cache and adds a custom in memory image cache, everything else doesn't get cached with that setting enabled.

We're not sure if that cache location is used for the http cache or not but it's worth a try.

**Reply by gtg250y** [reply]
Switched over and it seems to be working fine, now nothing gets added to this cache folder. I don't remember what RAM usage was before, but it doesn't look like it's growing indefinitely or anything so far. Getting more youtube throttled speed loads, but I feel like that was already happening lately (probably a battle that invidious/yt-dl have to always fight).

I should note that switching (requires a restart) doesn't nuke the existing cache in this folder, because apparently that timing is upon starting freetube when not in that experimental mode (i.e. you have to have that http cache option turned off, fully close, then open freetube to clear this cache, then turn on that setting and restart again).

**Reply by absidue** [reply]
When you enable that setting FreeTube tells Elctron every time it starts to disable the HTTP cache. So the disk cache not getting deleted with that setting enabled is probably because none of Electron's HTTP cache code gets run, even the deletion code.

The reason we have that setting is because Electron was causing lots of disk activity (writes and reads) while watching videos. Unfortunately there is no way to disable just the caching to the disk, disabling the HTTP cache also disables it in memory :(. So we wrote our own in memory image cache, that gets enabled when you enable that setting, it's not perfect and errors randomly so that's why the setting is still experimental.
